libr/core/cmd_print.c

```diff
@@ -3333,6 +3333,67 @@ static bool cmd_print_blocks(RCore *core, const char *input) {
 	return result;
 }
 
+static bool cmd_pse(RCore *core, const char *input) {
+	if (input[2] == ' ') {
+		const char *fname = r_str_trim_head_ro (input + 3);
```

Comment:
All this logic should be moved into a separate function. the cmd_pse should be a dispatcher with if/switch case:fcn

Reply:
The logic from where to where please?

Reply:
all the code inside the if (' ') { .. thisstuff .. }` ina separate function. dont make long functions, try to make the code more functional and easy to read. for example this code that is messing with paths, concat strings and such can be moved into a separate function char *tmp = resolve_encoding_path(fname); and return null if there's no such encoding available in home or system paths.